dashbuilder-shared/dashbuilder-dataset-shared/src/test/java/org/dashbuilder/dataset/engine/function/AggregateFunctionsTest.java

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.dashbuilder.dataset.engine.function;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.Before;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+public class AggregateFunctionsTest {
+
+    DateFormat DATE_FORMAT = new SimpleDateFormat("dd-MM-yyyy HH:mm:ss");
+    List<String> STRING_LIST = Arrays.asList("a", "b", "c");
+    List<String> STRING_LIST_WITH_NULL = Arrays.asList("a", "b", "c", null);
+    List<Date> DATE_LIST = new ArrayList<>();
+    List<Date> DATE_LIST_WITH_NULL = new ArrayList<>();
+    List<Double> NUMERIC_LIST = Arrays.asList(1d, 2d, 3d);
+    List<Double> NUMERIC_LIST_WITH_NULL = Arrays.asList(1d, 2d, 3d, null);
+    Date date1;
```

Comment:
Nice coverage! But there is one edge case which wasn't taken into consideration: what if ALL values in the list are null? In such case the aggregation functions call `sorted.get(0)` which would throw `IndexOutOfBoundException`.

I think this should be covered in aggregation functions implementation as well (e.g. imagine someone adds new values to a DB column when doing migration and sets all cells to null). Could you please cover that case as well?